services/horizon/internal/flags.go

```diff
@@ -809,16 +742,52 @@ func setCaptiveCoreConfiguration(config *Config, options ApplyOptions) error {
 		}
 	}
 
+	var defaultNetworkConfig networkConfig
 	if config.Network != "" {
-		err := createCaptiveCoreConfigFromNetwork(config)
+		var err error
+		defaultNetworkConfig, err = getCaptiveCoreConfigFromNetworkParameter(config)
 		if err != nil {
 			return err
 		}
-	} else {
-		err := createCaptiveCoreConfigFromParameters(config, options)
+		config.NetworkPassphrase = defaultNetworkConfig.NetworkPassphrase
+		config.HistoryArchiveURLs = defaultNetworkConfig.HistoryArchiveURLs
+	}
+
+	if config.NetworkPassphrase == "" {
+		return fmt.Errorf("%s must be set", NetworkPassphraseFlagName)
+	}
+
+	if len(config.HistoryArchiveURLs) == 0 {
+		return fmt.Errorf("%s must be set", HistoryArchiveURLsFlagName)
+	}
```

Comment:
```suggestion
	}  else {
		if config.NetworkPassphrase == "" {
			return fmt.Errorf("%s must be set", NetworkPassphraseFlagName)
		}

		if len(config.HistoryArchiveURLs) == 0 {
			return fmt.Errorf("%s must be set", HistoryArchiveURLsFlagName)
		}
	}
```

the check on `config.NetworkPassphrase` and `config.HistoryArchiveURLs` only needs to applied if `config.Network` is empty